### PR TITLE
Add request headers and retry failed operations

### PR DESCRIPTION
**Pull request by marcinwyszynski**
## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -3,9 +3,7 @@ builds:
       - CGO_ENABLED=0
       - GOPRIVATE="github.com/spacelift-io"
     mod_timestamp: '{{ .CommitTimestamp }}'
-    flags:
-      - -trimpath
-    ldflags: ['-s -w']
```
Comment (michieldewilde) [inline .goreleaser.yml:8]: ldflags removed because of what reason?

**Review by michieldewilde — APPROVED** [review]

**Review by cube2222 — APPROVED** [review]

**Review by b4k3r — APPROVED** [review]

File: .goreleaser.yml (hunk 1 above)
Comment (marcinwyszynski) [reply .goreleaser.yml:8]: Because goreleaser provides a richer set by default, and I'm using those defaults.

Comment (marcinwyszynski) [reply .goreleaser.yml:8]: As per docs:

```
    # Default is `-s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}} -X main.builtBy=goreleaser`.
```

Comment (michieldewilde) [reply .goreleaser.yml:8]: OK

**Pull request merged**